Src/Newtonsoft.Json/JsonTextWriter.cs

```diff
@@ -317,27 +319,28 @@ private void UpdateCharEscapeFlags()
         /// </summary>
         protected override void WriteIndent()
         {
-            _writer.WriteLine();
-
             // levels of indentation multiplied by the indent count
             int currentIndentCount = Top * _indentation;
 
-            if (currentIndentCount > 0)
-            {
-                if (_indentChars == null)
-                {
-                    _indentChars = new string(_indentChar, 10).ToCharArray();
-                }
+            int newLineLen = SetIndentChars();
 
-                while (currentIndentCount > 0)
-                {
-                    int writeCount = Math.Min(currentIndentCount, 10);
+            _writer.Write(_indentChars, 0, newLineLen + Math.Min(currentIndentCount, 12));
 
-                    _writer.Write(_indentChars, 0, writeCount);
+            while ((currentIndentCount -= 12) > 0)
+            {
+                _writer.Write(_indentChars, newLineLen, Math.Min(currentIndentCount, 12));
+            }
+        }
 
-                    currentIndentCount -= writeCount;
-                }
+        private int SetIndentChars()
+        {
+            string writerNewLine = _writer.NewLine;
+            if (_newLine != writerNewLine)
+            {
+                _indentChars = ((_newLine = writerNewLine) + new string(_indentChar, 12)).ToCharArray();
```

Comment:
Ah. I see. It'd be a more expensive check, especially since 99% of the time the exact same strings would be in place and string `==` is optimised to short-circuit to true on same objects, but it's probably worth it on balance. Done.